packages/TorneloScoresheet/src/hooks/appMode/recordingState.ts

```diff
@@ -359,21 +398,58 @@ export const makeUseRecordingState =
       });
     };
 
+    const positionPress = (
+      position: Position,
+      promotion: PieceType | undefined,
+    ) => {
+      // If there's no currently selected from square - assert that the pressed position has a piece
+      if (
+        !pressToMoveSelectedFromSquare &&
+        !appModeState.board.find(p => p.position === position)?.piece
+      ) {
+        return;
+      }
+      if (!pressToMoveSelectedFromSquare) {
+        setPressToMoveSelectedFromSquare({
+          position,
+          colour: colours.lightYellow,
+        });
+        return;
+      }
+      // The case where the user presses the same square twice -
+      // we clear the squares
+      if (position === pressToMoveSelectedFromSquare.position) {
+        setPressToMoveSelectedFromSquare(undefined);
+        return;
+      }
+      move(
+        {
+          from: pressToMoveSelectedFromSquare.position,
+          to: position,
+        },
+        promotion,
+      );
+      setPressToMoveSelectedFromSquare(undefined);
+    };
```

Comment:
It also looks like you're not using the context you created before? Instead we are putting this logic in the recording state,
Wouldn't it be better to store this logic in its own context? 
I think it would be better for 2 reasons:
- So that we do not make the recording state's larger than it should be
- This click to move logic is a front end consideration, whereas the recording state should be storing business logic related to the view model (the recording state)
- It means you can resuse this functionality wiithout having to copy code (for example in the edit move state you've had to repeat all these functions, if they are stored in a separate context, you wouldnt have to, you'd only need to pass through the move function into the context)